iree/compiler/Dialect/Flow/Transforms/Passes.cpp

```diff
@@ -55,6 +55,18 @@ static llvm::cl::opt<bool> clEnableConvToImg2Col(
     llvm::cl::desc("Enable converting convolution ops to img2col form."),
     llvm::cl::init(false));
 
+static llvm::cl::opt<bool> clEnablePaddingLinalgOps(
```

Comment:
would love to hear @MaheshRavishankar's comments on this - having this here is quite inflexible - we need something that will compose with the dynamic target pipelines he's been adding (and in general, move away from adding any flags here that are related to codegen)

Reply:
@benvanik I am also not a big fan of flags but I can't see other alternatives. We need this padding to be materialized at the flow level not sure the CPU dynamic target pipeline can do anything at this point.
 IMO padding at this level (at least for matmul) is globally optimal and we maybe just remove it if GPU backend is good with that ?

Reply:
One thing to try here is to run the `buildkite:benchmark` (or whatever that label is called) to see if all pipelines benefit from it (or at least they dont regress). Then we can drop the command line and set the option to 4.

In general though, having these LLVM sytle options is the worst thing we can do. At least to start with can we move to using https://mlir.llvm.org/docs/PassManagement/#instance-specific-pass-options . There were issues with global initialization order, but that should be fixed now. @asaadaldien since you are touching this part, do you mind moving the flags to that at least.

@benvanik this transformation needs to happen at flow level before allocation. I think we should just pick a reasonable baseline and say all backends do this (pick something like 4) and avoid having this flag. WDYT?

Reply:
@MaheshRavishankar to clarify moving the flags to `PassPipelineOptions` ?

Reply:
The question now how do we configure/pass these options when we do `iree-translate?` ?

Reply:
This needs a cleanup. At the very least all of these need to be moved into pipeline options, and then the benchmark should be updated to not use `-iree-mlir-to-vm-bytecode-module`. Instead it needs to use different pipelines and control the options through that. Ill get around to this in short order.